keras_cv/models/video_classification/vivit.py

```diff
@@ -0,0 +1,201 @@
+
+    # Train model
+    model.compile(
+        optimizer="adam",
+        loss="sparse_categorical_crossentropy",
+        metrics=[
+            keras.metrics.SparseCategoricalAccuracy(name="accuracy"),
+        ],
+    )
+
+    model.fit(frames, labels, epochs=3)
+
+    ```
+    """
+
+    def __init__(
+        self,
+        tubelet_embedder,
+        positional_encoder,
+        inp_shape,
+        num_classes,
+        transformer_layers=8,
+        num_heads=8,
+        embed_dim=128,
+        layer_norm_eps=1e-6,
+        **kwargs,
+    ):
+        if not isinstance(tubelet_embedder, keras.layers.Layer):
+            raise ValueError(
+                "Argument `tubelet_embedder` must be a "
+                " `keras.layers.Layer` instance "
+                f" . Received instead "
+                f"tubelet_embedder={tubelet_embedder} "
+                f"(of type {type(tubelet_embedder)})."
+            )
+
+        if not isinstance(positional_encoder, keras.layers.Layer):
+            raise ValueError(
+                "Argument `positional_encoder` must be a "
+                "`keras.layers.Layer` instance "
+                f" . Received instead "
+                f"positional_encoder={positional_encoder} "
+                f"(of type {type(positional_encoder)})."
+            )
+
+        inputs = keras.layers.Input(shape=inp_shape)
+        patches = tubelet_embedder(inputs)
+        encoded_patches = positional_encoder(patches)
+
+        for _ in range(transformer_layers):
+            x1 = keras.layers.LayerNormalization(epsilon=1e-6)(encoded_patches)
+            attention_output = keras.layers.MultiHeadAttention(
+                num_heads=num_heads,
+                key_dim=embed_dim // num_heads,
+                dropout=0.1,
+            )(x1, x1)
+
+            x2 = keras.layers.Add()([attention_output, encoded_patches])
+
+            x3 = keras.layers.LayerNormalization(epsilon=1e-6)(x2)
+            x3 = keras.Sequential(
+                [
+                    keras.layers.Dense(
+                        units=embed_dim * 4, activation=keras.ops.gelu
+                    ),
+                    keras.layers.Dense(
+                        units=embed_dim, activation=keras.ops.gelu
+                    ),
+                ]
+            )(x3)
+
+            encoded_patches = keras.layers.Add()([x3, x2])
+
+        representation = keras.layers.LayerNormalization(
+            epsilon=layer_norm_eps
+        )(encoded_patches)
+        representation = keras.layers.GlobalAvgPool1D()(representation)
+
+        outputs = keras.layers.Dense(units=num_classes, activation="softmax")(
+            representation
+        )
+
+        super().__init__(inputs=inputs, outputs=outputs, **kwargs)
+
+        self.inp_shape = inp_shape
+        self.num_heads = num_heads
+        self.num_classes = num_classes
+        self.tubelet_embedder = tubelet_embedder
+        self.positional_encoder = positional_encoder
+
```

Comment:
Please add a build method here and build all the layers defined in init (with trainable variable) with their corresponding input shape.